### ISSUE TITLE
Cannot upgrade react to 17 thanks to npm complaining that react-codemirror2 supports react 16 only

### ISSUE THREAD

**Issue by vass-david** [scniro/react-codemirror2#271]
Hi!

I can see that react version in this repo is correct (up to 17.x), but for some reason my npm complains that it's only up to 16.x. Any idea what went wrong? I've tried cleaning my npm cache, but it didn't help (I didn't expect it will anyway). Might it be that current version published in npm has an outdated package.json?

Thanks!

**Comment by MikeHingley22** [comment]
The latest release (https://github.com/scniro/react-codemirror2/tree/7.1.0) has the following in the package.json file : 

https://github.com/scniro/react-codemirror2/blob/cc3a47d0707b060bf1a1e42ccb258b9b7c0882fd/package.json#L56

I think the reason is that npm is reading the peer dependancy of 16.x and enforcing that.  Latest version on npm seems to be 7.2.1 and that also has the same 16.x upper restriction.

**Comment by AlexCriollo** [comment]
> The latest release (https://github.com/scniro/react-codemirror2/tree/7.1.0) has the following in the package.json file :
> 
> https://github.com/scniro/react-codemirror2/blob/cc3a47d0707b060bf1a1e42ccb258b9b7c0882fd/package.json#L56
> 
> I think the reason is that npm is reading the peer dependancy of 16.x and enforcing that. Latest version on npm seems to be 7.2.1 and that also has the same 16.x upper restriction.

i see that last release was back on 2020. Any word on when is the next one?

**Comment by iiiusky** [comment]
ping

**Comment by ksathyanm** [comment]
@elsigh @ihollander @fongandrew @anandsainath @scniro ping

**Comment by vldmr1986** [comment]
Same for React  18

**Comment by supunkavinda** [comment]
Ended up copying [index.tsx](https://github.com/scniro/react-codemirror2/blob/master/src/index.tsx) to the project :(

**Comment by av-k** [comment]
As a temporary solution can be used the next approach.

Add to `package.json` the next:
```
"scripts": {
  ...
  "post-install": "node bin/post-install-worker",
  "postinstall": "npm run post-install"
},
  ```
  
  `post-install-worker` file contains:
```
const {readFileSync, writeFileSync} = require('fs');
const path = require('path');

const fixPeerPeerDependenciesForReactCodeMirror = () => {
  const pkgJsonPath = 'node_modules/react-codemirror2/package.json';
  const pkgJsonBuffer = readFileSync(path.join(__dirname, '..', pkgJsonPath));
  const pkgJson = JSON.parse(pkgJsonBuffer);
  const pkgJsonPeerDepsKeys = Object.keys(pkgJson.peerDependencies || {});

  if (pkgJsonPeerDepsKeys.length > 0) {
    pkgJson.peerDependencies = pkgJsonPeerDepsKeys.reduce((acc, depKey) => {
      if (depKey === 'react') {
        return acc;
      }
      return ({
        ...acc,
        [depKey]: pkgJson.peerDependencies[depKey],
      });
    }, {});

    writeFileSync(pkgJsonPath, JSON.stringify(pkgJson, null, 2));
  }
};

fixPeerPeerDependenciesForReactCodeMirror();
```
  
 So after `postinstall` run of npm hook the file `package.json` will be corrected.

**Comment by s100** [comment]
React 17 support was added under #224, which was published in `react-codemirror2@7.3.0`. Super work!
